lib/mongo/protocol/serializers.rb

```diff
@@ -265,7 +265,7 @@ def self.serialize(buffer, value, max_bson_size = nil, validating_keys = BSON::C
           #
           # @since 2.5.0
           def self.deserialize(buffer)
-            BSON::Document.from_bson(buffer)
+            BSON::Document.from_bson(buffer, mode: :bson)
```

Comment:
The problem was that we were sending commands to mongocryptd, which adds encryption placeholders, and then we were deserializing the responses without `mode: :bson`, so any int64s were coming back as regular integers.